src/axom/primal/geometry/Polygon.hpp

```diff
@@ -22,63 +22,184 @@ namespace axom
 {
 namespace primal
 {
+namespace
+{
+static constexpr int DEFAULT_MAX_NUM_VERTICES = 20;
+
+/*!
+ * \brief A helper class for the Polygon class
+ *
+ * This class helps define a StackArray with a current size and additional
+ * functions to mirror axom::Array.
+ */
+template <typename T, int N>
+struct StaticArray : public StackArray<T, N>
+{
+  AXOM_HOST_DEVICE int size() const { return m_size; }
+  AXOM_HOST_DEVICE void push_back(const T& obj)
+  {
+    SLIC_ASSERT(m_size + 1 <= N);
+    StackArray<T, N>::m_data[m_size++] = obj;
+  }
+
+  AXOM_HOST_DEVICE void clear()
+  {
+    for(T& datum : StackArray<T, N>::m_data)
+    {
+      datum = T();
+    }
+    m_size = 0;
+  }
+
+private:
+  int m_size {0};
+};
```

Comment:
Would it make sense to move this struct to `axom::core`? (Perhaps in a follow-up PR)

Reply:
> Would it make sense to move this struct to axom::core? (Perhaps in a follow-up PR)

I think it would be useful, generalizable for other uses. 
I'll revisit this in the follow-up PR to this one, adding polygon clipping on device (details may change in the process).